web_api.py

```diff
@@ -0,0 +1,290 @@
+from datetime import datetime
+import os
+import json
+import uuid
+from typing import Dict, Union
+
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+from db.db_engine import Upload, User, UploadStatus, save_to_json, get_engine
+from flask import Flask, request, jsonify, Response, redirect, url_for, flash
+
+from explainer import OUTPUTS_FOLDER
+
+app = Flask(__name__)
+UPLOAD_FOLDER = 'uploads'
+OUTPUT_FOLDER = 'outputs'
+app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
+app.config['OUTPUT_FOLDER'] = OUTPUT_FOLDER
+
+
+@app.route('/upload', methods=['POST'])
+def upload():
+    """
+    API endpoint for uploading a file.
+
+    Returns:
+        flask.Response: JSON response containing the UID of the upload.
+    """
+    if not os.path.isdir(app.config['UPLOAD_FOLDER']):
+        os.mkdir(app.config['UPLOAD_FOLDER'])
+
+    if 'file' not in request.files:
+        return jsonify({'error': 'No file provided.'}), 400
+
+    file = request.files['file']
+    uid = generate_uid()
+    email = request.form.get('email')
+    engine = get_engine()
+    with Session(engine) as session:
+        f_upload = Upload(uid=uid, filename=os.path.basename(file.filename),
+                          upload_time=datetime.now(), status='pending')
+        new_filename = f_upload.upload_path(app.config['UPLOAD_FOLDER'])
+        session.add(f_upload)
+        session.commit()
+
+        if email:
+            user = _get_user_by_email(email, session)
+            if not user:
+                user = User(email=email)
+            user.uploads.append(f_upload)
+            f_upload.user = user
+            f_upload.user_id = user.id
+            session.add_all([user, f_upload])
+            session.commit()
+
+    file.save(new_filename)
+    return jsonify({'uid': str(uid)})
+
+    # filename = get_filename(file.filename)
+    # timestamp = datetime.now().strftime('%Y-%m-%d-%H-%M-%S')  # int(time.time())
+    # email = request.form.get('email')
+    # if email:
+    #     uid = save_upload_with_email(file, email)
+    # else:
+    #     new_filename = f"{filename}_{timestamp}_{uid}"
+    #     file_path = os.path.join(app.config['UPLOAD_FOLDER'], new_filename)
+    #     file.save(file_path)
+    # response = {'uid': uid}
+    # return jsonify(response)
+
+
+@app.route('/status/<uid>', methods=['GET'])
+def status(uid):
```

Comment:
Separate code to small functions.  Read more: https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be